### PR TITLE
Reset password link should be present in all pages #1403

### PR DESCRIPTION
**Pull request by Anand-Royy**
## Describe your changes
Reset password link is added to all the login pages.

## Issue number
#1403 

## Please ensure all items are checked off before requesting a review:

- [x] I deployed the code locally.
- [x] I have performed a self-review of my code.
- [x] I have included the issue # in the PR.
- [x] I didn't use any hardcoded values (otherwise it will not scale, and will make it difficult to maintain consistency across the application).
- [x] I have labelled the PR correctly.
- [x] My PR is granular and targeted to one specific feature.
- [x] I took a screenshot or a video and attached this to PR if there is a UI change.

## Screenshots

image

image

### CODE REVIEW

**Comment by gorkem-bwl** [comment]
Can you please send a screenshot showing those pages?

**Comment by gorkem-bwl** [comment]
> Sure
> 
Great, thanks. I believe the light theme looks ok on your end, right? Can you check that too? 

For my info - there should have been a PR template when you issue this PR. Did you see it? I am not sure if that is in action. 

It should have looked something like this:

image

**Comment by Anand-Royy** [comment]
>Great, thanks. I believe the light theme looks ok on your end, right? Can you check that too?

Yes, the light theme is also fine. 

image

image

**Comment by Anand-Royy** [comment]
>For my info - there should have been a PR template when you issue this PR. Did you see it? I am not sure if that is in action.

Ok, I'll add it

**Comment by gorkem-bwl** [comment]
> > For my info - there should have been a PR template when you issue this PR. Did you see it? I am not sure if that is in action.
> 
> Ok, I'll add it

OK. It's important you do not remove it next time. It's meant to be filled in properly.

**Comment by Anand-Royy** [comment]
> OK. It's important you do not remove it next time. It's meant to be filled in properly.

I have added it. Please check.

File: Client/src/Pages/Auth/Login.jsx [hunk 1]
```diff
@@ -206,6 +244,32 @@ const StepOne = ({ form, errors, onSubmit, onChange, onBack }) => {
...
+					<Typography
+						component="span"
+						color={theme.palette.primary.main}
+						ml={theme.spacing(2)}
+						sx={{ userSelect: "none" }}
+						onClick={handleNavigate}
+					>
+						Reset password
+					</Typography>
+				</Box>
```
[the remark this answers is not in the document]
Comment (Samsonbtc247) [reply Client/src/Pages/Auth/Login.jsx:272]: > _:bulb: Codebase verification_
> 
> _:hammer_and_wrench: Refactor suggestion_
> 
> **Yo! The review's on point - this Box needs some serious refactoring!**
> 
> Looking at the code, I found the same Box component with identical styling being used in three places within Login.jsx (in LandingPage, StepOne, and StepTwo components). The Box contains the same "Forgot password?" and "Reset password" text with identical positioning and styling.
> 
> - `Client/src/Pages/Auth/Login.jsx`: The Box component is duplicated in:
>   - LandingPage component (around line 247)
>   - StepOne component (around line 419)
>   - StepTwo component (around line 544)
> 
> The suggested ResetPasswordLink component would indeed help eliminate this duplication and improve accessibility with the added keyboard navigation support.
> 
> 
> 
> <details>
> <summary>🔗 Analysis chain</summary>
> 
> **Yo! Extract this Box component to reduce duplication**
> 
> This Box component is copy-pasted in multiple places. Let's keep it DRY!
> 
> ```diff
> +const ResetPasswordLink = ({ handleNavigate }) => {
> +  const theme = useTheme();
> +  return (
> +    <Box
> +      textAlign="center"
> +      sx={{
> +        position: "absolute",
> +        bottom: 0,
> +        left: "50%",
> +        transform: `translate(-50%, 150%)`,
> +      }}
> +    >
> +      <Typography
> +        className="forgot-p"
> +        display="inline-block"
> +        color={theme.palette.primary.main}
> +      >
> +        Forgot password?
> +      </Typography>
> +      <Typography
> +        component="span"
> +        color={theme.palette.primary.main}
> +        ml={theme.spacing(2)}
> +        sx={{ 
> +          userSelect: "none",
> +          cursor: "pointer",
> +          "&:focus": {
> +            outline: `2px solid ${theme.palette.primary.main}`,
> +            outlineOffset: "2px"
> +          }
> +        }}
> +        role="link"
> +        tabIndex={0}
> +        onKeyPress={(e) => e.key === 'Enter' && handleNavigate()}
> +        onClick={handleNavigate}
> +      >
> +        Reset password
> +      </Typography>
> +    </Box>
> +  );
> +};
> 
> +ResetPasswordLink.propTypes = {
> +  handleNavigate: PropTypes.func.isRequired,
> +};
> ```
> 
> </details>
> 
> <details>
> <summary>🏁 Scripts executed</summary>
> 
> The following scripts were executed for the analysis:
> 
> Script:
> 
> ```shell
> #!/bin/bash
> # Check for similar Box components in other files
> rg -l 'Forgot password.*Reset password' --type jsx
> 
> ```
> 
> Length of output: 80
> 
> ---
> 
> Script:
> 
> ```shell
> #!/bin/bash
> # Let's try with a different approach to find similar Box components
> rg -U "Box.*position.*absolute.*bottom.*left.*transform.*translate.*Forgot password" --type ts --type tsx --type js --type jsx
> 
> ```
> 
> Length of output: 156
> 
> ---
> 
> Script:
> 
> ```shell
> #!/bin/bash
> # Search for files containing both "Forgot password" and "Reset password"
> rg -l "Forgot password" --type ts --type tsx --type js --type jsx
> 
> ```
> 
> Length of output: 95
> 
> ---
> 
> Script:
> 
> ```shell
> #!/bin/bash
> # Let's also check for any Box components with similar positioning
> rg -U "Box.*position.*absolute.*bottom.*left.*transform.*translate" --type ts --type tsx --type js --type jsx
> 
> ```
> 
> Length of output: 139
> 
> ---
> 
> Script:
> 
> ```shell
> #!/bin/bash
> # Let's try without file type restrictions
> rg -U "Box.*position.*absolute.*bottom.*left.*transform.*translate.*Forgot password"
> 
> ```
> 
> Length of output: 86
> 
> ---
> 
> Script:
> 
> ```shell
> #!/bin/bash
> # Search for files containing both phrases without type restrictions
> rg -l "Forgot password"
> 
> ```
> 
> Length of output: 171
> 
> ---
> 
> Script:
> 
> ```shell
> #!/bin/bash
> # Find all potential React/JS/TS files
> fd -e js -e jsx -e ts -e tsx
> 
> ```
> 
> Length of output: 9378
> 
> ---
> 
> Script:
> 
> ```shell
> #!/bin/bash
> # Let's check the content of ForgotPassword.jsx and Login.jsx for similar Box components
> cat "Client/src/Pages/Auth/ForgotPassword.jsx" "Client/src/Pages/Auth/Login.jsx"
> 
> ```
> 
> Length of output: 20642
> 
> ---
> 
> Script:
> 
> ```shell
> #!/bin/bash
> # Let's also check if there are any other similar Box components with absolute positioning
> rg -U "Box.*textAlign.*position.*absolute" Client/src/Pages/Auth/
> 
> ```
> 
> Length of output: 67
> 
> </details>
> 
> 

#1403

**Review by ajhollid — CHANGES_REQUESTED** [review]
Hi @Anand-Royy ,

Thanks for your contribution here!

A recent PR has moved the `Login.jsx` file to `/Auth/Login/Login.jsx`, the `/Auth/Login.jsx` page no longer exists.

Could you please either merge `develop` into your feature branch and move your changes to `/Auth/Login/Login.jsx` or just close this PR and move the changes to a new branch from develop?

Thanks!

**Comment by ajhollid** [comment]
It looks like you've just copied your `Login.jsx` and pasted it over the most recent version of `/Auth/Login/Login.jsx`.  

That will overwrrite all the changes made in recent PRs.

I'm going to close this PR as this is not mergeable, if you'd like to commit the code for adding the reset password link the steps to follow are

1.  Checkout `develop` remote branch
2. Create a new branch from the current state of `develop`
3. Make your changes, commit, and push.

**Comment by Anand-Royy** [comment]
Hi @ajhollid, I pulled the changes from the `develop` branch and I could find the `/Auth/Login/Login.jsx` file. But when I make changes to this file or the files inside the `/Auth/Login/Component`, I can't see any changes being reflected to the frontend. 

Also I can still see the previous `login.jsx` file outside the `Login` folder and when I make changes to this file then I can see the changes getting reflected on the frontend too. Is this supposed to happen?

**Comment by ajhollid** [comment]
@Anand-Royy I can't really troubleshoot your repository for you remotely, my suggestion to quickly get back on track is just reclone the repository.

Any other Git troubleshooting you'll have to handle on your end.

**Comment by Anand-Royy** [comment]
>A recent PR has moved the Login.jsx file to /Auth/Login/Login.jsx, the /Auth/Login.jsx page no longer exists.

@ajhollid I can see both the files in the repo

image

**Comment by Anand-Royy** [comment]
@ajhollid Yes, so the changes made to the file `/Auth/Login.jsx` is actually reflecting to the frontend and not the file changes made in `/Auth/Login/Login.jsx`. Please check this because my local repo is updated with the latest commit in the remote branch and I can observe these changes.

**Comment by ajhollid** [comment]
Hi @Anand-Royy ,

You are correct, the old page was being rendered by the router.  I have removed the old page and corrected the route, you can make a new PR with your changes in the `/Auth/Login/Login.jsx` page if you'd like.

Thanks!

**Pull request closed without merging**